File: web/src/lib/generation/slopGuard.ts
import { callAiService } from "@/lib/aiService/client";

// Mirrors ai-service/app/schemas/slop_guard.py
export type SlopGuardVerdict = "pass" | "soft_nudge" | "hard_reject";

export interface SlopGuardResult {
  verdict: SlopGuardVerdict;
  reason: string;
}

export async function checkSlopGuard(
  rawText: string,
): Promise<SlopGuardResult> {
  return callAiService<SlopGuardResult>("/slop-guard", { raw_text: rawText });
}

File: ai-service/app/schemas/slop_guard.py
from enum import StrEnum

from pydantic import BaseModel, Field


class SlopGuardVerdict(StrEnum):
    PASS = "pass"
    SOFT_NUDGE = "soft_nudge"
    HARD_REJECT = "hard_reject"


class SlopGuardRequest(BaseModel):
    raw_text: str = Field(min_length=1, max_length=20_000)


class SlopGuardResult(BaseModel):
    verdict: SlopGuardVerdict
    reason: str
